ethtx/providers/semantic_providers/semantics_repository.py

```diff
@@ -254,6 +254,9 @@ def get_semantics(self, chain_id: str, address: str) -> Optional[AddressSemantic
         # amend semantics with locally stored updates
         amend_contract_semantics(address_semantics.contract)
 
+        if self._records is not None:
+            self._records.append(address)
```

Comment:
If an error occurs before the `end_record()` method, how `self._records` will be cleared?

Reply:
self._record is replaced with every decode so it shouldn't be a problem